src/components/SQForm/SQFormDatePicker.tsx

```diff
@@ -16,6 +21,32 @@ const useStyles = makeStyles(() => ({
   }
 }));
 
+interface SQFormDatePickerProps extends BaseFieldProps {
+  /** Disabled property to disable the input if true */
+  isDisabled?: boolean;
+  /** Required property used to highlight input and label if not fulfilled */
+  isRequired?: boolean;
+  /** Placeholder text used inside the input field to provide hints to the user */
+  placeholder?: string;
+  /** Custom onBlur event callback */
+  onBlur?: React.FocusEventHandler<HTMLInputElement>;
+  /** Custom onChange event callback */
+  onChange?: (
+    date: React.ChangeEvent<Element | moment.Moment> | unknown
+  ) => void;
+  /** Disable specific date(s) (day: DateIOType) => boolean
+   * This is a predicate function called for every day of the month
+   * Return true to disable that day or false to enable that day
+   */
+  setDisabledDate: (day: unknown) => boolean;
```

Comment:
do we want to avoid unknown?

Reply:
Nope, I tried to pass the `Moment` type to but it highlighted it as a warning. The default type is `unknown` suggested by linter. 
![image](https://user-images.githubusercontent.com/34863307/128022092-2ff9f2a6-136d-452e-a38c-b8ae21057113.png)

Reply:
Date should be of type `DateIOType` as seen here https://material-ui-pickers.dev/api/DatePicker
DateIOType is, according to the link above, "DateIOType — date object type of your linked date-io adapter (Moment, DayJS, etc.)"

Is there _any_ way we can get more specific than `unknown`?

Reply:
Yes, they have mentioned `DateIOType ` in docs. but material-UI `picker` is using the `shouldDisableDate?: (day: unknown) => boolean;` type under the hood
![image](https://user-images.githubusercontent.com/34863307/128031104-ba61a042-80af-44c2-841c-c1b2d0968616.png)

**Refrence file path:**  node_modules\@material-ui\pickers\_helpers\date-utils.d.ts

Reply:
This prop should be optional